Deploy to Netlify

ColeKettler opened the pull request:
## Overview

Establishes a `netlify.toml` config to deploy the `master`, `develop`, and PR branches to Netlify.

## Testing Instructions

- Confirm that the [Netlify deploy preview build](https://app.netlify.com/sites/districtbuilder/deploys/5ef36279926191000755929c) passes.
- Go to the [deploy preview](https://5ef36279926191000755929c--districtbuilder.netlify.app/) and confirm that a default Jenkins site loads with no content security policy errors in the console.

Resolves #3

hectcastro commented on netlify.toml:
```diff
@@ -0,0 +1,28 @@
+[Settings]
+  ID = "publicmapping"
```
I think that the ID should be something like `districtbuilder` since this is expected to be the `districtbuilder.org` marketing website.

ColeKettler replied:
I renamed the `districtbuilder` site we had in Netlify that was connected to `azavea/districtbuilder-2`. It's now `districtbuilder-prototype`.

hectcastro commented on netlify.toml:
```diff
@@ -0,0 +1,28 @@
+  [headers.values]
+    Content-Security-Policy = """
+      default-src 'self';
+      script-src 'self' 'unsafe-inline';
```
Are there currently any script includes that require `unsafe-inline`? If not, I think it would be good to start with it off and selectively add it as we need to.

ColeKettler replied:
Whatever default inline scripts are being used by Jekyll were throwing a CSP error. We should revisit including this when we build out our first version of the UI.

ColeKettler commented:
I'm absolutely in favor of moving towards a trunk-based development model like [GitHub Flow](https://guides.github.com/introduction/flow/) here. I think Netlify works really well with that model and it sheds some complexity.

I think that's an appropriate decision to make within the context of this PR. I've been working with Aaron on this so I can communicate the decision we make. A few preliminary questions:

- Is there benefit in maintaining semantic versioning tags as part of the release process? If we don't, then...
- ...how do we structure the `CHANGELOG`? We may be able to eschew its use and rely on PRs and merge commits as documentation, as recommended by GitHub Flow.

ColeKettler commented:
I'm absolutely in favor of moving towards a trunk-based development model like [GitHub Flow](https://guides.github.com/introduction/flow/) here. I think Netlify works really well with that model and it sheds some complexity.

ColeKettler commented:
Cool, I've:

- created a `main` branch that's identical to `develop` and set it to be the default branch
- removed the `develop` branch
- retargeted this PR to `main`
- removed the branch deploy context from the Netlify config
- removed the `develop` branch from the Netlify branch deploy list

This leaves the `master` branch intact to preserve the GH Pages redirect to the DistrictBuilder site. When we're ready for an initial release of this marketing site, we can either delete the `master` branch or rename it to `gh-pages` or something if we need to preserve the redirect.

ColeKettler commented:
#8 has been created to keep track of removing / renaming the `master` branch.
